src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -277,6 +291,54 @@ private IList<ContentItem> FindItemsImplementation(PatternSet definition, IEnume
             return itemsList;
         }
 
+        internal string GetRelatedFileExtensionProperty(string assemblyPath, IEnumerable<Asset> assets)
+        {
+            //E.g. if path is "lib/net472/A.B.C.dll", the prefix will be "lib/net472/A.B.C."
+            string assemblyPrefix = assemblyPath.Substring(0, assemblyPath.LastIndexOf('.') + 1);
+
+            if (_assemblyRelatedExtensions.TryGetValue(assemblyPrefix, out string relatedProperty))
+            {
+                return relatedProperty;
+            }
+
+            List<string> relatedFileExtensionList = null;
+            foreach (Asset asset in assets)
+            {
+                if (asset.Path is not null)
+                {
+                    string extension = Path.GetExtension(asset.Path);
+                    if (extension != string.Empty &&
+                        //Assembly properties are files with extensions ".dll", ".winmd", ".exe", see ManagedCodeConventions.
+                        !extension.Equals(".dll", StringComparison.OrdinalIgnoreCase) &&
```

Comment:
Regarding the case sensitivity, I'd recommend validating based on the current NuGet behavior.

Example: Are `System.DLL` and `System.dll` treated equivalently. 
What aobut `System.dll` and `SYSTEM.dll`.